### PR TITLE
fix: one click binding postgress sorting issue when primary key is no…

### PR DESCRIPTION
**Pull request by sbalaji1192**
…t available

## Description
When the primary key is not present in a postgres table, we should update generated query to not have its reference.

#### PR fixes following issue(s)
Fixes https://github.com/appsmithorg/appsmith/issues/24079

#### Type of change

- Bug fix (non-breaking change which fixes an issue)

- [ ] Manual
- [x] Jest
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [x] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by laveena-en** [comment]
@sbalaji1192 On choosing a postgres table where primary key isn’t present, the query generated for Save, has an undefined reference which is causing the save functionality to fail cc: @chandannkumar 
image

File: app/client/src/WidgetQueryGenerators/PostgreSQL/index.ts [hunk 1]
```diff
@@ -35,14 +35,22 @@ export default abstract class PostgreSQL extends BaseQueryGenerator {
...
+              4: `"{{${orderBy} || '${formConfig.primaryColumn}'}}"`,
+              5: `{{${sortOrder} ? "" : "DESC"}}`,
+            },
+          }
+        : {
+            isValuePresent: orderBy,
+            template: "$4",
+            params: {
+              4: `{{${orderBy} ? "ORDER BY " + ${orderBy} + "  " + (${sortOrder} ? "" : "DESC") : ""}}`,
+            },
```
Comment (vsvamsi1) [inline app/client/src/WidgetQueryGenerators/PostgreSQL/index.ts:52]: Minor thing 
```
{
  isValuePresent: orderBy,
        ...(formConfig.primaryColumn
          ? {
              template: `ORDER BY $4 $5`,
              params: {
                4: `"{{${orderBy} || '${formConfig.primaryColumn}'}}"`,
                5: `{{${sortOrder} ? "" : "DESC"}}`,
              },
            }
          : {
              template: "$4",
              params: {
                4: `{{${orderBy} ? "ORDER BY " + ${orderBy} + "  " + (${sortOrder} ? "" : "DESC") : ""}}`,
              },
            }),
            
      }
```

Comment (sbalaji1192) [reply app/client/src/WidgetQueryGenerators/PostgreSQL/index.ts:52]: @vsvamsi1 Do you see any benefits of this style than the one I used?

**Comment by laveena-en** [comment]
Verified the fix and callouts for:

- One click binding's functionality when the primary key is not present in a postgres table.

LGTM

**Review by vsvamsi1 — APPROVED** [review]

File: app/client/src/WidgetQueryGenerators/PostgreSQL/index.ts (hunk 1 above)
Comment (vsvamsi1) [reply app/client/src/WidgetQueryGenerators/PostgreSQL/index.ts:52]: I felt isValuePresent is the common property which isn't changing hence we can apply the spread on the other two properties  template and params. We can leave this comment its not too important.

**Review by vsvamsi1 — APPROVED** [review]

**Pull request merged**